src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor_code.go

```diff
@@ -240,16 +242,38 @@ func (a *StoragePowerActorCode_I) Surprise(rt Runtime) {
 // Method utility functions
 ////////////////////////////////////////////////////////////////////////////////
 
-func (st *StoragePowerActorState_I) _rtGetPowerEntryOrAbort(rt Runtime, minerAddr addr.Address) PowerTableEntry {
+func (a *StoragePowerActorCode_I) _rtGetPowerEntryOrAbort(rt Runtime, minerAddr addr.Address) PowerTableEntry {
+	h, st := a.State(rt)
 	powerEntry, found := st.PowerTable()[minerAddr]
 
 	if !found {
-		rt.AbortStateMsg("sm._rtGetPowerEntryOrAbort: miner not found in power table.")
+		rt.AbortStateMsg("spa._rtGetPowerEntryOrAbort: miner not found in power table.")
 	}
 
+	Release(rt, h, st)
 	return powerEntry
 }
 
+func (a *StoragePowerActorCode_I) _rtGetPledgeCollateralReqForMinerOrAbort(rt Runtime, minerAddr addr.Address) actor.TokenAmount {
+	inds := rt.CurrIndices()
+	h, st := a.State(rt)
+
+	activePower, inactivePower, ok := st._getPowerTotalForMiner(minerAddr)
+	if !ok {
+		rt.AbortArgMsg("spa._rtGetPledgeCollateralReqForMinerOrAbort: miner not in PowerTable.")
+	}
+
+	currPledge, ok := st._getCurrPledgeForMiner(minerAddr)
+	if !ok {
+		rt.AbortArgMsg("spa._rtGetPledgeCollateralReqForMinerOrAbort: miner not in EscrowTable.")
+	}
+
+	minBalance := inds.BlockReward_PledgeCollateralReq(activePower, inactivePower, currPledge)
```

Comment:
Why is the current pledge a parameter to the pure function that computes the pledge requirement?